### PR TITLE
feat(): Added functionality to update profile page

### PR DESCRIPTION
**Pull request by dlabrecq**
Added functionality to link GitHub and OpenShift accounts.
Added a button to fetch/link the user's GitHub avatar image URL.
Added a new ngx-widgets component to count remaining characters for user's bio.
Added Bootstrap switch (jw-bootstrap-switch-ng2) component to toggle profile properties.
Added collapsible panel to display user's token.
Added CopyService to copy the user's token to system clipboard.
Added company field not shown in screenshot.

Layout and features are based on latest mockup from @jryhanyc:
https://redhat.invisionapp.com/share/3ZBFATW28#/screens

Reviewed with @catrobson

Notes:

1. This PR depends on ngx-widgets https://github.com/fabric8-ui/ngx-widgets/pull/77

2. This PR depends on ngx-login-client https://github.com/fabric8-ui/ngx-login-client/issues/27

3. The user's profile is updated successfully; however, the UserService does not refresh its users. Therefore, the UI will not reflect changes because the previously obtained user is now stale (i.e., unless you login, again). See issue https://github.com/fabric8-ui/ngx-login-client/issues/57

4. Privacy properties are not yet supported by the /api/users API. Therefore, the toggle switches are currently shown disabled. See issue https://github.com/almighty/almighty-core/issues/1288

5. If an account is removed via Keycloak, the UI still shows the accounts as linked because the tokens are cached via localStorage. GitHub and OpenShift tokens are only refreshed every 10+ minutes. See issue https://github.com/fabric8-ui/ngx-login-client/issues/58

6. When linking GitHub and OpenShift accounts, the user is currently redirected to the getting started page. This is because the UI doesn't support refreshing the current page without generating an error. See issue https://github.com/fabric8io/fabric8-ui/issues/1264

7. Not using Patternfly's pficon-arrow icon because fabric8-ui does not have the latest Patternfly css and fonts. Instead, I'm using a similar icon from font awesome for the "Link Profile Photo" button. See issue https://github.com/fabric8io/fabric8-ui/issues/474

Screenshot:
profile

### CODE REVIEW

File: src/app/create/codebases/services/github.service.ts [hunk 1]
```diff
@@ -221,12 +223,47 @@ export class GitHubService implements OnDestroy {
     }
   }
 
+<<<<<<< a406c55bdabd321bcdf72558a95af4e23bb06040
```
Comment (ppitonak) [inline src/app/create/codebases/services/github.service.ts]: This shouldn't be here

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 5a5747c16a7cb1e7444ba29d8536ea96fdc9bd44
  * Your subject line is longer than 50 characters

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

**Comment by fabric8cd** [comment]
@dlabrecq snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1263-fabric8-ui.int.rdu2c.fabric8.io

File: src/app/profile/update/update.component.scss [hunk 2]
```diff
@@ -1,10 +1,80 @@
...
 .checkbox input[type="checkbox"] {
   margin-left: 0;
   margin-right: 5px;
   position: relative;
 }
-.fields-status-pf { margin-top: -5px; }
-.getting-started { padding-bottom: 20px; }
-.margin-top-40 { margin-top: 40px; }
+
+.fields-status-pf {
```
Comment (joshuawilson) [inline src/app/profile/update/update.component.scss]: Please make these single attributes into one line, it is better for debugging.

Comment (dlabrecq) [inline src/app/profile/update/update.component.scss]: Will do. However, it doesn't make for good readability. In the event of an unlikely error, it's only one line number off.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: ae468cbb692459f5c3e49c8a5c0edddff11e439a
  * Your subject line is longer than 50 characters

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

File: src/app/profile/update/update.component.scss (hunk 2 above)
Comment (dlabrecq) [inline src/app/profile/update/update.component.scss]: Done. However, it doesn't make for good readability. In the event of an unlikely error, the debugger is only one line number off.

An error here would be no different than the numerous other selectors with multiple lines.

**Comment by fabric8cd** [comment]
@dlabrecq snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1263-fabric8-ui.int.rdu2c.fabric8.io

**Pull request merged**